src/EnergyPlus/PlantManager.cc

```diff
@@ -889,22 +887,27 @@ namespace PlantManager {
 							this_comp.TypeOf_Num = TypeOf_Pipe;
 							this_comp.GeneralEquipType = GenEquipTypes_Pipe;
 							this_comp.CurOpSchemeType = NoControlOpSchemeType;
+							this_comp.compPtr = Pipes::LocalPipeData::factory( TypeOf_Pipe, CompNames( CompNum ) );
```

Comment:
Somewhere inside the code, we need to have a switch block that checks the _type_ of object and interprets the idf data based on that type, right?  So even if we had a generic "factory" that you just passed in a bunch of idf text, it would have to create a chiller-specific data structure instance, for example, right?  This seems like the right place for it, given what I am trying to achieve at this point.  Looking beyond this initial refactoring I'm sure there are major other improvements, and we should certainly discuss them.  But I think this change should be suitable for making it better without painting us into a corner.